### ISSUE TITLE
`touch` does not work for directories

### ISSUE THREAD

**Issue by kik4444** [nushell/nushell#11757]
### Describe the bug

Today I tried doing this to restore my timestamp on a directory after accidentally creating a file there
```nushell
touch -r original_dir copied_dir
```
But got an error `Failed to create file: Is a directory (os error 21)`

After some more tests I've come to believe that `touch` just doesn't work on directories, `--reference` or not.

### How to reproduce

See above

### Expected behavior

I expected this to work like with the coreutils `touch`
```nushell
^touch -r original_dir copied_dir # works
```

### Configuration

| key                | value                                                            |
| ------------------ | ---------------------------------------------------------------- |
| version            | 0.89.0                                                           |
| branch             | makepkg                                                          |
| commit_hash        | 2c1560e281d65a50d90b30c42006755b364b4928                         |
| build_os           | linux-x86_64                                                     |
| build_target       | x86_64-unknown-linux-gnu                                         |
| rust_version       | rustc 1.75.0 (82e1608df 2023-12-21) (Arch Linux rust 1:1.75.0-1) |
| cargo_version      | cargo 1.75.0                                                     |
| build_time         | 2024-01-09 23:05:05 +00:00                                       |
| build_rust_channel | release                                                          |
| allocator          | mimalloc                                                         |
| features           | dataframe, default, extra, sqlite, trash, which, zip             |
| installed_plugins  | query, query json, query web, query xml                          |

**Comment by kik4444** [comment]
Looking at the code `src/filesystem/touch.rs` it seems like the issue happens around here
```rust
for (index, item) in vec![target].into_iter().chain(rest).enumerate() {
...

    if let Err(err) = OpenOptions::new().write(true).create(true).open(&item) {
        return Err(ShellError::CreateNotPossible {
            msg: format!("Failed to create file: {err}"),
            span: call
                .positional_nth(index)
                .expect("already checked positional")
                .span,
        });
    };
...
```
For every path given to the `touch` command, Nu first tries to create a file by that name. Afterwards it changes the mtime or atime on the created file. But if the `target` is an existing directory then `OpenOptions::new().write(true).create(true).open(&item)` will fail

**Comment by fdncred** [comment]
It seems like it should check if it a file or a folder somewhere.

**Comment by kik4444** [comment]
> It seems like it should check if it a file or a folder somewhere.

I made a commit simply not to create a file if the path is a dir. Unfortunately I can't test it because I've hit another bug. I built the latest `main` branch but I get an error when I try to run `touch -r nu_plugin_query.d nu.d`
```nushell
Error: nu::shell::cant_convert

  × Can't convert to bool.
   ╭─[entry #1:1:1]
 1 │ touch -r nu_plugin_query.d nu.d
   ·          ────────┬────────
   ·                  ╰── can't convert string to bool
   ╰────
```
This is unrelated to my changes. I'm trying to track down when this happened.

**Comment by kik4444** [comment]
Alright I'll make a separate PR the `touch` issue though since it seems like an easy fix

**Comment by kik4444** [comment]
> Feel free to ping the PR author to see if they have time to investigate.

Would making a comment in https://github.com/nushell/nushell/pull/11492 suffice? I feel like this deserves its own issue...

**Comment by sholderbach** [comment]
closed by #12005
